encoding/csv/stream.ts

```diff
@@ -0,0 +1,51 @@
+import { Buffer, BufReader } from "../../io/buffer.ts";
```

Comment:
Now, this is going to be the problem: any of the new streams based APIs are not allowed to depend on any Deno.Reader/Writer APIs. You'll have to wait for #1970 to land

Reply:
@uki00a #1970 landed, so this should be now doable. `BufReader` & `BufWriter` dont exist anymore now (as they dont make sense for web streams). However, i see that csv also depends on textproto, which hasnt been ported yet

Reply:
Rewrote `CSVStream` not to depend on `Deno.Reader` in [c573f87](https://github.com/denoland/deno_std/pull/1993/commits/c573f876d6bbf7e85e915b1b0c724594f0495ef9) :slightly_smiling_face: